scapy/contrib/metawatch.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (C) 2019 Brandon Ewing <brandon.ewing@warningg.com>
+#               2019 Guillaume Valadon <guillaume.valadon@netatmo.com>
+##
+# This program is published under a GPLv2 license
+
+# scapy.contrib.description = Arista Metawatch
+# scapy.contrib.status = loads
+
+from types import MethodType
+
+from scapy.layers.l2 import Ether
+from scapy.fields import ByteField, ShortField, FlagsField, SecondsIntField, \
+    UTCTimeField
+from scapy.error import Scapy_Exception
+
+
+class TrailerBytes(bytes):
+    """
+    Reverses slice operations to take from the back of the packet,
+    not the front
+    """
```

Comment:
Wouldn't simply reversing the field order fix this?

Reply:
I have no idea =)
I blindly copy the original PR as I don't know the protocol.

Reply:
It's been a while since I was in the headspace when I wrote this, so I apologize if this explanation isn't clear.  

Metamako (now Arista) created an ethernet trailer specification described here: https://lists.sandelman.ca/pipermail/tcpdump-workers/2018-July/001053.html

It follows the original frame FCS on the wire, and can optionally contain TLVs in addition to the fixed fields.  Since this technically is part of the Ethernet frame, and not part of the ethernet payload, my believe is that it has to be processed before completely consuming the payload bytes.  In order to do that, bytes have to be removed from the end of the packet, hence this class.

A similar approach is taken in the Wireshark dissector for the trailer as well: https://github.com/wireshark/wireshark/blob/master/epan/dissectors/packet-metamako.c

Reply:
@gpotter2 FCSField only works with `unpack()` valid formats. This implementation is more generic.

Reply:
Fair enough. In that case i'd like the two implementations to be merged